src/examples/show.scala

```diff
@@ -66,13 +70,21 @@ trait GenericShow[Out] extends AutoDerivation[[X] =>> Show[Out, X]] {
   override def split[T](ctx: SealedTrait[Typeclass, T]): Show[Out, T] =
     (value: T) =>
       ctx.choose(value) { sub =>
-        val anns =
-          sub.annotations.filterNot(_.isInstanceOf[scala.SerialVersionUID])
+        val anns = filterUnwantedAnns(
+          (sub.annotations ++ sub.inheritedAnnotations).distinct
+        )
+
         val annotationStr =
           if (anns.isEmpty) "" else anns.mkString("{", ",", "}")
 
         prefix(annotationStr, sub.typeclass.show(sub.value))
       }
+
+  def filterUnwantedAnns(anns: IArray[Any]): IArray[Any] =
+    anns
+      .filterNot(_.isInstanceOf[scala.SerialVersionUID])
+      .filterNot(_.isInstanceOf[scala.annotation.showAsInfix])
```

Comment:
shouldn't we filter out these annotations from `inheritedAnns` in the first place?

Reply:
yep, after filtering out `Object` and `scala*` we don't need to filter these annotations